app/utils/get_filtered_records.py

```diff
@@ -159,23 +160,38 @@ def should_include(d, k, v):
     else:
         result = query_dicts
 
-    def date_filter(record, start_date=None, end_date=None):
+    def date_filter(record, start_date=None, end_date=None, sampling_date_type=True):
         status = True
 
         if start_date is not None:
-            status = status and record["sampling_end_date"] and start_date <= record["sampling_end_date"]
+            # If sampling_date_type is True, compare to the sampling start date
```

Comment:
i think this would be cleaner:

```
record_end = record["sampling_end_date"] if sampling_date_type else record["publication_date"]
```

Then we can basically keep the old logic, just using `record_end` in place of `record["sampling_end_date"]`

Reply:
Also we should make the param name more bool like, maybe `use_sampling_date`